src/App.js

```diff
@@ -1,85 +1,151 @@
-          placeholder="What needs to be done?"
-        />
-      </header>
-
-      <section className="main">
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">asdfghj</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">qwertyuio</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">zxcvbnm</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-4" />
-              <label htmlFor="todo-4">1234567890</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+class App extends React.Component {
+  state = {
+    todoList: [],
+    originalTodoList: [],
+    idCounter: 0,
+    filterIdentifier: 'all',
+  };
+
+  handleAddTodo = (title) => {
+    this.setState(prevState => ({
+      originalTodoList: [
+        ...prevState.originalTodoList,
+        {
+          id: prevState.idCounter + 1,
+          title,
+          isCompleted: false,
+        },
+      ],
+
+      idCounter: prevState.idCounter + 1,
+    }));
+
+    this.filterTodoList();
+  };
+
+  filterTodoList = () => {
+    this.setState((prevState) => {
+      const { filterIdentifier, originalTodoList } = prevState;
+
+      switch (filterIdentifier) {
+        case 'active':
+          return ({
+            todoList: originalTodoList.filter(todo => !todo.isCompleted),
+          });
+        case 'completed':
+          return ({
+            todoList: originalTodoList.filter(todo => todo.isCompleted),
+          });
+        default:
+          return ({
+            todoList: [...originalTodoList],
+          });
+      }
+    });
+  };
+
+  toggleAllTodosCompleted = () => {
+    this.setState(prevState => ({
+      originalTodoList: prevState.originalTodoList.map(todo => ({
+        ...todo,
+        isCompleted: prevState.originalTodoList.some(t => !t.isCompleted),
```

Comment:
```suggestion
        isCompleted: prevState.originalTodoList.some(todo => !todo.isCompleted),
```
Не сокращай имена переменных